test/OpenTelemetry.Instrumentation.Dependencies.Tests/HttpWebRequestTests.Basic.net461.cs

```diff
@@ -76,28 +79,26 @@ public async Task HttpDependenciesInstrumentationInjectsHeadersAsync()
             parent.TraceStateString = "k1=v1,k2=v2";
             parent.ActivityTraceFlags = ActivityTraceFlags.Recorded;
 
-            using (new HttpWebRequestInstrumentation(tracer, new HttpClientInstrumentationOptions()))
-            {
-                using var response = await request.GetResponseAsync();
-            }
+            using var response = await request.GetResponseAsync();
 
-            Assert.Equal(2, spanProcessor.Invocations.Count); // begin and end was called
-            var span = (SpanData)spanProcessor.Invocations[1].Arguments[0];
+            Assert.Equal(2, activityProcessor.Invocations.Count); // begin and end was called
+            var activity = (Activity)activityProcessor.Invocations[1].Arguments[0];
 
-            Assert.Equal(parent.TraceId, span.Context.TraceId);
-            Assert.Equal(parent.SpanId, span.ParentSpanId);
-            Assert.NotEqual(parent.SpanId, span.Context.SpanId);
-            Assert.NotEqual(default, span.Context.SpanId);
+            Assert.Equal(parent.TraceId, activity.Context.TraceId);
+            Assert.Equal(parent.SpanId, activity.ParentSpanId);
+            Assert.NotEqual(parent.SpanId, activity.Context.SpanId);
+            Assert.NotEqual(default, activity.Context.SpanId);
 
             string traceparent = request.Headers.Get("traceparent");
             string tracestate = request.Headers.Get("tracestate");
 
-            Assert.Equal($"00-{span.Context.TraceId}-{span.Context.SpanId}-01", traceparent);
+            Assert.Equal($"00-{activity.Context.TraceId}-{activity.Context.SpanId}-01", traceparent);
             Assert.Equal("k1=v1,k2=v2", tracestate);
 
             parent.Stop();
         }
 
+        /* TBD: ActivitySource doesn't support custom format TraceIds.
```

Comment:
I couldn't find a way to make this test work. AFAIK there isn't a way on Activity/Source API to inject a custom TraceId mechanism. I think @MikeGoldsmith already brought that up. /cc @tarekgh @cijothomas

Reply:
what is exactly the ask here? do you want to create a custom trace Id? is there any more information what is expect to be valid custom trace id? I am just asking to ensure tracking the request.

---
In reply to: [429571008](https://github.com/open-telemetry/opentelemetry-dotnet/pull/694#discussion_r429571008) [](ancestors = 429571008)

Reply:
OpenTelemetry [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/237ab5fb7e412e0f2ea170b6904640951e0a876d/specification/trace/api.md#spancontext) regarding TraceID says needs to support 16-byte array with at least one non-zero byte to be deemed valid.

What are the limitations of the TraceId from the `Activity` side?

Reply:
I saw a comment somewhere that we needed to support generating Ids in different formats, for example [B3](https://github.com/openzipkin/b3-propagation). I can't seem to find it now.

Reply:
Spec says SpanContext conforms to W3CTraceContext spec. ActivityContext is conforming to W3C spec as well. I don't think there is anything needed from ActivityContext.

The propogation (extraction, injection) of context - is not the concern of Activity( or Span).